### PR TITLE
[results-processor] Provide GitHub token for api.github.com

### PR DESCRIPTION
**Pull request by gsnedders**
Sometimes (e.g., from GitHub Actions), we might get passed a URL to process which on api.github.com, which requires a token to access. Thus, we should pull down our GitHub token and add that as a header.

See also https://github.com/web-platform-tests/wpt.fyi/issues/4020, which this hopefully fixes.

### CODE REVIEW

**Review by jcscottiii — APPROVED** [review]
This is my first time really digging into the processor code.

LGTM with some nits.

File: results-processor/processor.py [hunk 1]
```diff
@@ -112,6 +112,14 @@ def known_extension(path):
                 return e
         return None
 
+    def _secret(self, token_name):
+        key = self.datastore.key('Token', token_name)
```
[the remark this answers is not in the document]
Comment (gsnedders) [reply results-processor/processor.py:116]: Figuring out how to pass that through to the processor (which doesn't inherently rely on Flask/main/etc. today) seems kinda difficult. We can certainly add logging, though?

[the remark this answers is not in the document]
Comment (jcscottiii) [reply results-processor/processor.py:120]: Can you add a follow up issue for this? We can address that part later.

Something around logging which uploader is accessing the secret.

After you add the issue, feel free to merge.

File: results-processor/processor.py [hunk 2]
```diff
@@ -123,15 +135,31 @@ def _download_gcs(self, gcs):
 
     def _download_http(self, url):
         assert url.startswith('http://') or url.startswith('https://')
-        _log.debug("Downloading %s", url)
+        _log.debug('Downloading %s', url)
+        extra_headers = None
+        if urlsplit(url).hostname == GITHUB_API_HOSTNAME:
```
Comment (jcscottiii) [inline results-processor/processor.py:140]:
```suggestion
        if urlsplit(url).hostname == self.GITHUB_API_HOSTNAME:
```

Comment (jcscottiii) [reply results-processor/processor.py:140]: This should fix the lint error

File: results-processor/processor.py [hunk 3]
```diff
@@ -123,15 +135,31 @@ def _download_gcs(self, gcs):
...
+            extra_headers = {
+                'Authorization': 'Bearer ' + self._github_token,
+                'X-GitHub-Api-Version': GITHUB_API_VERSION,
```
Comment (jcscottiii) [inline results-processor/processor.py:143]:
```suggestion
                'X-GitHub-Api-Version': self.GITHUB_API_VERSION,
```

This should fix the lint error.

File: results-processor/processor.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (gsnedders) [reply results-processor/processor.py:120]: Filed https://github.com/web-platform-tests/wpt.fyi/issues/4044

**Comment by gsnedders** [comment]
I haven't had the time to investigate, but it looks like the web_components_test CI job is just constantly failing now?

**Comment by gsnedders** [comment]
> I haven't had the time to investigate, but it looks like the web_components_test CI job is just constantly failing now?

Filed https://github.com/web-platform-tests/wpt.fyi/issues/4046 for this; I'm just going to land this despite the failing check because it's pre-existing.

**Comment by gsnedders** [comment]
https://staging.wpt.fyi/status seems to show everything getting stuck as WPTFYI_PROCESSING, though not showing up as empty any more. But probably time to go back to #4020 to analyse what's going on?

**Pull request merged**